Musicca/app/src/main/java/com/example/musicca/adapters/CurrentPlaylistAdapter.java

```diff
@@ -50,83 +58,203 @@ public CurrentPlaylistAdapter(Context context, List<String> songObjectIds, Strin
     @NonNull
     @Override
     public CurrentPlaylistAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
-        View view = LayoutInflater.from(context).inflate(R.layout.item_queue_song, parent, false);
+        View view = LayoutInflater.from(context).inflate(R.layout.item_playlist_song, parent, false);
         return new ViewHolder(view);
     }
 
     @Override
     public void onBindViewHolder(@NonNull CurrentPlaylistAdapter.ViewHolder holder, int position) {
         String songObjectId = songObjectIds.get(position);
         //Song song = songs.get(position);
-        holder.bind(songObjectId);
+        holder.bind(songObjectId, position);
     }
 
     @Override
     public int getItemCount() {
         return songObjectIds.size();
     }
 
-    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
-        ImageView ivAlbum;
-        TextView tvTitle;
-        TextView tvArtist;
+    class ViewHolder extends RecyclerView.ViewHolder {
+        public ImageView ivAlbum;
+        public TextView tvTitle;
+        public TextView tvArtist;
+        public ImageView ivLike;
+        public TextView tvLikes;
 
         public ViewHolder(@NonNull View itemView) {
             super(itemView);
             ivAlbum = itemView.findViewById(R.id.ivAlbum);
             tvTitle = itemView.findViewById(R.id.tvTitle);
             tvArtist = itemView.findViewById(R.id.tvArtist);
-            itemView.setOnClickListener(this);
+            ivLike = itemView.findViewById(R.id.ivLike);
+            tvLikes = itemView.findViewById(R.id.tvLikes);
+
+            // Usage: override double tap, single tap and long press
```

Comment:
For first-time users, how to teach them to use these gestures? Consider adding clues in your app.

Reply:
I added a toast message to notify users about using a double tap: https://github.com/gracewainaina/Musicca/blob/like-songs/Screenshots/toastmessage_doubletap.mp4